pandas/core/groupby.py

```diff
@@ -468,6 +473,7 @@ def _set_selection_from_grouper(self):
 
             if len(groupers):
                 self._group_selection = ax.difference(Index(groupers)).tolist()
+                self._reset_cache('_selected_obj')
```

Comment:
I am not sure, I think `_reset_cache` is only necessary if `_group_selection` changes, this is why I ordered it in this way.